File: RepoTransBench/target_projects/C/Python/Wirtos_async2.h/public_tests/test_async2_api.py
import pytest
from src.async2 import AsyncArr, async_arr_push, async_arr_splice, async_arr_destroy

def test_public_async2_api():
    arr = AsyncArr()
    arr.count = 0
    arr.data = []
    # Push different numbers than private (e.g., 5, 16, 25)
    async_arr_push(arr, 5)
    async_arr_push(arr, 16)
    async_arr_push(arr, 25)

    assert arr.count == 3
    assert arr.data[0] == 5
    assert arr.data[1] == 16
    assert arr.data[2] == 25

    # Splice at a different index (remove second element)
    async_arr_splice(arr, 1, 1)
    assert arr.count == 2
    assert arr.data[0] == 5
    assert arr.data[1] == 25

    # Further splice (remove first element)
    async_arr_splice(arr, 0, 1)
    assert arr.count == 1
    assert arr.data[0] == 25

    # Test destroy resets to zero
    async_arr_destroy(arr)
    assert arr.count == 0

File: RepoTransBench/target_projects/C/Rust/Wirtos_async2.h/public_tests/test_async2_api.rs
use async2::AsyncArr;

#[test]
fn public_test_async2_api() {
    let mut arr = AsyncArr {
        count: 0,
        values: [0; 10],
    };

    // Push different numbers than private (e.g., 5, 16, 25)
    arr.push(5);
    arr.push(16);
    arr.push(25);

    assert_eq!(arr.count, 3);
    assert_eq!(arr.values[0], 5);
    assert_eq!(arr.values[1], 16);
    assert_eq!(arr.values[2], 25);

    // Splice at a different index (remove second element)
    arr.splice(1, 1);

    assert_eq!(arr.count, 2);
    assert_eq!(arr.values[0], 5);
    assert_eq!(arr.values[1], 25);

    // Further splice (remove first element)
    arr.splice(0, 1);
    assert_eq!(arr.count, 1);
    assert_eq!(arr.values[0], 25);

    // Test destroy resets to zero
    arr.destroy();
    assert_eq!(arr.count, 0);
}